Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -58,42 +73,105 @@
     }
 
     STDMETHOD_(void, AddLines)(_In_reads_(pointsCount) CONST D2D1_POINT_2F* points, UINT32 pointsCount) {
+        _delayedEndFigureCall();
         m_geometrySink->AddLines(points, pointsCount);
         m_lastPoint = points[pointsCount - 1];
     }
 
     STDMETHOD_(void, AddBeziers)(_In_reads_(beziersCount) CONST D2D1_BEZIER_SEGMENT* beziers, UINT32 beziersCount) {
+        _delayedEndFigureCall();
         m_geometrySink->AddBeziers(beziers, beziersCount);
         m_lastPoint = beziers[beziersCount - 1].point3;
     }
 
-    STDMETHOD_(void, BeginFigure)(D2D1_POINT_2F startPoint, D2D1_FIGURE_BEGIN figureBegin) {
+    STDMETHOD_(void, AddLine)(D2D1_POINT_2F point) {
+        _delayedEndFigureCall();
+        m_geometrySink->AddLine(point);
+        m_lastPoint = point;
+    }
+
+    STDMETHOD_(void, AddBezier)(_In_ CONST D2D1_BEZIER_SEGMENT* bezier) {
+        _delayedEndFigureCall();
+        m_geometrySink->AddBezier(bezier);
+        m_lastPoint = bezier->point3;
+    }
+
+    STDMETHOD_(void, AddQuadraticBezier)(_In_ CONST D2D1_QUADRATIC_BEZIER_SEGMENT* bezier) {
+        _delayedEndFigureCall();
+        m_geometrySink->AddQuadraticBezier(bezier);
+        m_lastPoint = bezier->point2;
+    }
+
+    STDMETHOD_(void, AddQuadraticBeziers)(_In_reads_(beziersCount) CONST D2D1_QUADRATIC_BEZIER_SEGMENT* beziers, UINT32 beziersCount) {
+        _delayedEndFigureCall();
+        m_geometrySink->AddQuadraticBeziers(beziers, beziersCount);
+        m_lastPoint = beziers[beziersCount - 1].point2;
+    }
+
+    STDMETHOD_(void, AddArc)(_In_ CONST D2D1_ARC_SEGMENT* arc) {
+        _delayedEndFigureCall();
+        m_geometrySink->AddArc(arc);
+        m_lastPoint = arc->point;
+    }
+
+    STDMETHOD_(void, SetAllowsFigureCalls)(bool allows) {
```

Comment:
These methods that are not part of the ID2D1 geometry sink should not be STDMETHOD (rather they dont have to be)